src/Microsoft.AspNetCore.OData/Routing/Conventions/ODataRoutingConventions.cs

```diff
@@ -77,6 +77,7 @@ public static IList<IODataRoutingConvention> CreateDefault()
                 new RefRoutingConvention(),
                 new ActionRoutingConvention(),
                 new FunctionRoutingConvention(),
+                new NestedPathsRoutingConvention(),
```

Comment:
@habbes With reference to your comment on the PR description that the `NestedPathsRoutingConvention` should appear later in the list, what would happen if its promoted to close to the top of the list? For instance, if the user opts out of configuring the routing in their service using the `CreateDefault`, what risk do they face if this routing convention appears before `EntitySetRoutingConvention` or `EntityRoutingConvention` for example? If mis-ordering can occasion any risk can we catch that as the service is bootstrapping and throw an exception?

Reply:
Consider the following controller:
```c#
class CustomersController {
   Get(int key) () { / ... }
  
  GetOrders(int key) () {}

  [EnableNestedPaths]
   Get() {}

}
```

In the current setup of routing conventions, this is how the following paths would be routed:

Path    |    Action  | Convention
-------|------------|-------------
/Customers | `Get()`  | `EntitySetConvention`
/Customers(1) | `Get(int key)` | `EntityConvention`
/Customers(1)/Orders | `GetOrders(int key)` | `NavigationPropertyConvention`
/Customers(1)/Name  | `Get()` | `NestedPathsConvention`
/Customers(1)/Orders(2)/Name | `Get()` | `NestedPathsConvention`

So the "standard" conventions take precedence. Meaning the `NestedPathsConvention` is not triggered if there's a user-defined method created to handle that specific path.

If the conventions are miss-ordered, such that the `NestedPathsConvention` appears above the "standard" conventions, then all applicable paths will be routed to the `Get()` method (if it has `[EnableNestedPaths]`) even if there are other user-defined methods that could handle such paths:

Path    |    Action  | Convention
-------|------------|-------------
/Customers | `Get()`  | `NestedPathsConvention`
/Customers(1) | `Get()` | `NestedPathsConvention`
/Customers(1)/Orders | `Get()` | `NestedPathsConvention`
/Customers(1)/Name  | `Get()` | `NestedPathsConvention`
/Customers(1)/Orders(2)/Name | `Get()` | `NestedPathsConvention`

So the danger is that the user would not be able to override the `Get()` method with `[EnableNestedPaths]`.

I think this should be documented, but I'm not sure if it should go as far as throwing an exception. The same concern would follow if the user miss-ordered the attribute routing convention, making it appear later in the list. I think the user should be aware of the consequences, but not necessarily get an exception.